tests/test_projectbuilder.py

```diff
@@ -111,6 +129,14 @@ def test_init(mocker, test_flit_path, legacy_path, test_no_permission, test_bad_
         build.ProjectBuilder(test_bad_syntax_path)
 
 
+def test_python_executable(test_flit_path):
```

Comment:
Do you want to test here passing in bytes too though, not? And explicit Unicode?